code/datums/components/speed_curse.dm

```diff
@@ -0,0 +1,38 @@
+/datum/component/speed_curse
+    var/mob/holder
+    var/slowdown
+    var/change_delay = 0
+    var/lasttime = 0
+
+/datum/component/speed_curse/Initialize(_change_delay)
+    _change_delay = (3 SECONDS)
+    change_delay = _change_delay
+
+/datum/component/speed_curse/RegisterWithParent()
+    RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/on_equip)
+    RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/on_drop)
+
+//Called when we are equipped to a user
+/datum/component/speed_curse/proc/on_equip(datum/source, mob/equipper, slot)
+    holder = equipper
```

Comment:
You shouldn't need to store a holder, just grab the current holder each time you need it since you can assume you'll always have one while processing

Reply:
that's wasteful, it's better to trust that equip and dropped are called, and just remember who you're equipped to.

otherwise the component has to get the parent loc and check if it's a mob every single time

strongly disagree

Reply:
@ninjanomnom 
Would you still like me to go ahead with this change, or keep it as is?

Reply:
You don't need to check it's a mob every time since as we've both said you can trust equipped or dropped was called. It doesn't matter too much in this case I suppose though either way, just trying to keep the same pattern in regards to avoiding extra refs in components that aren't needed.